numpy/ma/tests/test_core.py

```diff
@@ -1327,14 +1327,32 @@ def test_eq_on_structured(self):
         test = (a == a)
         assert_equal(test, [True, True])
         assert_equal(test.mask, [False, False])
+        test = (a == a[0])
+        assert_equal(test, [True, False])
+        assert_equal(test.mask, [False, False])
         b = array([(1, 1), (2, 2)], mask=[(1, 0), (0, 0)], dtype=ndtype)
         test = (a == b)
         assert_equal(test, [False, True])
         assert_equal(test.mask, [True, False])
+        test = (a[0] == b)
+        assert_equal(test, [False, False])
+        assert_equal(test.mask, [True, False])
         b = array([(1, 1), (2, 2)], mask=[(0, 1), (1, 0)], dtype=ndtype)
         test = (a == b)
-        assert_equal(test, [True, False])
+        assert_equal(test, [True, True])
         assert_equal(test.mask, [False, False])
+        ndtype = [('A', int), ('B', [('BA', int), ('BB', int)])]
+        a = array([(1, (1, 1)), (2, (2, 2))],
+                  mask=[(0, (1, 0)), (0, (0, 1))], dtype=ndtype)
+        # complicated dtype, 2-dimensional array.
+        ndtype = [('A', int), ('B', [('BA', int), ('BB', int)])]
```

Comment:
Can you add a test with subarray types? So `[('A', int), ('B', float, 3)]` or something?